libs/api/domains/university-of-iceland/src/lib/universityOfIceland.resolver.ts

```diff
@@ -0,0 +1,71 @@
+import { Args, Query, Resolver } from '@nestjs/graphql'
+
+import { Inject, UseGuards } from '@nestjs/common'
+import format from 'date-fns/format'
+import is from 'date-fns/locale/is'
+
+import {
+  CurrentUser,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  UniversityOfIcelandService,
+  NemandiGetLocaleEnum,
+  NemandiFerillFerillGetLocaleEnum,
+} from '@island.is/clients/university-of-iceland'
+import { ApiScope } from '@island.is/auth/scopes'
+import { StudentInfoModel } from './models/studentInfo.model'
+import { StudentInfoDetailModel } from './models/studentInfoDetail.model'
+import { GetStudentInfoDetailInput } from './dto/getStudentInfoDetail.input'
+import { GetStudentInfoInput } from './dto/getStudentInfo.input'
+import { DownloadServiceConfig } from '@island.is/nest/config'
+import type { ConfigType } from '@island.is/nest/config'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(ApiScope.internal)
+@Resolver()
+export class UniversityOfIcelandResolver {
+  constructor(
+    private universityOfIcelandApi: UniversityOfIcelandService,
+    @Inject(DownloadServiceConfig.KEY)
+    private readonly downloadServiceConfig: ConfigType<
+      typeof DownloadServiceConfig
+    >,
+  ) {}
+
+  @Query(() => StudentInfoModel)
+  async getStudentInfo(
+    @CurrentUser() user: User,
+    @Args('input') input: GetStudentInfoInput,
+  ): Promise<object> {
+    const data = await this.universityOfIcelandApi.studentInfo(
+      user,
+      input.locale as NemandiGetLocaleEnum,
+    )
+    return data
+  }
```

Comment:
Why not type `input.locale` as `NemandiGetLocaleEnum` instead of string to begin with?

Reply:
NestJS weirdness is making this change very unwieldy, revisiting this later